ipa-core/src/protocol/ipa_prf/shuffle/malicious.rs

```diff
@@ -223,16 +323,23 @@ async fn h3_verify<C: Context, S: BooleanArray>(
 ///
 /// ## Panics
 /// Panics when conversion from `BooleanArray` to `Vec<Gf32Bit` fails.
-fn compute_row_hash<S, I>(keys: &[StdArray<Gf32Bit, 1>], row_iterator: I) -> Hash
+fn compute_row_hash<S, B, I>(keys: &[StdArray<Gf32Bit, 1>], row_iterator: I) -> Hash
```

Comment:
The name `row_iterator` is used for two different things in this function -- an iterator over rows, and an iterator within a row. Maybe `rows_iterator` and `row_iterator`? Or `row_iterator` and `col_iterator`?

There might also be a better name for the  function. Maybe `compute_and_hash_tags` (contrast with `compute_and_add_tags`) or `hash_rows_and_tags`?

Another difference with `compute_and_add_tags` is whether or not the tags are secret shared. I don't think that has to be reflected in the name (although it could be), but maybe is worth calling out in a comment.